pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -250,3 +279,61 @@ func newTestHTTPClient(response *http.Response, err error) *http.Client {
 		},
 	}
 }
+
+type benchmarkingRoundTripper struct {
+	dataType benchmarkingDataType
+}
+
+func (brt benchmarkingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return makeProtoResponseForBenchmarking(http.StatusOK, brt.dataType, network.ProtoAcceptContent), nil
+}
+
+func newBenchmarkingtHTTPClient(dataType benchmarkingDataType) *http.Client {
+	return &http.Client{
+		Transport: benchmarkingRoundTripper{dataType: dataType},
+	}
+}
+
+type benchmarkingDataType int
+
+const (
+	NoData benchmarkingDataType = iota
+	SmallData
+	MediumData
+	LargeData
+)
```

Comment:
Should we simplify this by just passing `podName` into the `makeProto...` function?

Reply:
yes ok